actors/builtin/miner/miner_actor.go

```diff
@@ -151,42 +157,71 @@ func (a Actor) ControlAddresses(rt Runtime, _ *adt.EmptyValue) *GetControlAddres
 	rt.State().Readonly(&st)
 	info := getMinerInfo(rt, &st)
 	return &GetControlAddressesReturn{
-		Owner:  info.Owner,
-		Worker: info.Worker,
+		Owner:        info.Owner,
+		Worker:       info.Worker,
+		ControlAddrs: info.ControlAddresses,
 	}
 }
 
 type ChangeWorkerAddressParams struct {
-	NewWorker addr.Address
+	NewWorker       addr.Address
+	NewControlAddrs []addr.Address
 }
 
 func (a Actor) ChangeWorkerAddress(rt Runtime, params *ChangeWorkerAddressParams) *adt.EmptyValue {
 	var effectiveEpoch abi.ChainEpoch
-	worker := resolveWorkerAddress(rt, params.NewWorker)
+
+	worker := addr.Undef
+	if params.NewWorker != addr.Undef {
```

Comment:
This can never happen: `addr.Undef` does not serialize so the miner will have to specify their existing worker address.

Always resolve this, and check at 202 against the current worker address.

Reply:
This is done.